submissions/SergeyShytikov/js-oop-frogger-task/app.js

```diff
@@ -0,0 +1,125 @@
+const ROWS = 6;
+const COLUMNS = 5;
+const CORRELATION_INDEX = 1.25;
+const FIELD_WIDTH = 500;
+const FIELD_HEIGHT = FIELD_WIDTH;
+const CELL_WIDTH = FIELD_WIDTH / COLUMNS;
+const CELL_HEIGHT = Math.ceil(FIELD_HEIGHT / ROWS);
+const TOP_BORDER = Math.trunc(-CELL_HEIGHT / (CORRELATION_INDEX * 4));
+const X_START_POSITION_PLAYER = FIELD_WIDTH / (CORRELATION_INDEX * 2);
+const Y_START_POSITION_PLAYER = FIELD_HEIGHT / CORRELATION_INDEX;
+const X_START_POSITION_ENEMY = -CELL_WIDTH;
+const Y_START_POSITION_ENEMY = FIELD_HEIGHT / 2 - CORRELATION_INDEX * 16;
+const ENEMY_AVATAR = "images/enemy-bug.png";
+const PLAYER_AVATAR = "images/char-boy.png";
+
+const Character = function (x, y, sprite) {
+  this.x = x;
+  this.y = y;
+  this.sprite = sprite;
+};
+Character.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// --------------------------------------ENEMY------------------------
+const Enemy = function (x, y, speed, sprite, player) {
+  Character.call(this, x, y, sprite);
+  this.speed = speed;
+  this.player = player;
+};
+Enemy.prototype = Object.create(Character.prototype);
+Enemy.prototype.constructor = Enemy;
+Enemy.prototype.update = function (dt) {
+  this.x =
+    this.x > ctx.canvas.width
+      ? X_START_POSITION_ENEMY
+      : (this.x += this.speed * dt);
+  this.checkCollision();
+};
+Enemy.prototype.checkCollision = function () {
+  if (
+    Math.trunc(this.x + CELL_WIDTH / 2) >= this.player.x &&
+    Math.trunc(this.x) < this.player.x + CELL_WIDTH &&
+    this.y + 2 === this.player.y
+  ) {
+    this.player.resetPosition();
+  }
+};
+const allEnemies = [];
+const _allEnemies = [];
+let levelCounter = 0;
+
+function resetEnemies() {
+  if (levelCounter > 3) {
+    allEnemies.splice(0, allEnemies.length),
+      _allEnemies.splice(0, _allEnemies.length),
+      (levelCounter = 0);
```

Comment:
What this fragment of code does?
Why expressions are separated with commas?

Reply:
comma separation is a mistake, 
and the purpose of this code is clearing the arrays with the bugs when level 3 of difficulty is reached. 
The two arrays help me to randomize the generation of the bugs more and allow them to appear on top of each other